nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/persistence/FlowConfigurationDAO.java

```diff
@@ -25,11 +25,41 @@
 import org.apache.nifi.controller.UninheritableFlowException;
 import org.apache.nifi.controller.serialization.FlowSerializationException;
 import org.apache.nifi.controller.serialization.FlowSynchronizationException;
+import org.apache.nifi.encrypt.StringEncryptor;
+import org.apache.nifi.util.NiFiProperties;
 
 /**
  * Interface to define service methods for FlowController configuration.
  */
 public interface FlowConfigurationDAO {
+    /**
+     * Performs any initialization needed. This should be called only by the
+     * framework.
+     * @param encryptor protected property encryptor/decryptor
+     * @param nifiProperties collection of NiFi properties
+     * @throws IOException thrown if Flow Configuration fails to initialize
+     */
+    void initialize(StringEncryptor encryptor, NiFiProperties nifiProperties) throws IOException;
+
+    /**
+     * Locks the Flow to prevent cross-thread Writeing
+     */
+    void writeProtectFlow();
```

Comment:
i mentioned it in a previous comment but the need to make locking calls available by callers into this class indicates something a bit awkward happening.  See previous comments.